app/models/request.rb

```diff
@@ -144,4 +145,15 @@ def valid_uid?(uid, field)
         true
       end
     end
+
+    # If a request fails to be a approved we make sure there were not orphan
+    # project records left in our Rails database that do not have a matching
+    # project in Mediaflux (i.e. collection asset).
+    def cleanup_incomplete_project
+      project = Project.find_by_id(project_id)
+      if project && project.mediaflux_id.nil?
```

Comment:
Notice that we only delete the project if it does not have a matching project in Mediaflux